Docs/GitHub rate limits

kudos opened the pull request:
Adds docs for Github auth and rate limiting and removes the Travis CI configuration section that it supersedes. I moved the `JSPM_GITHUB_AUTH_TOKEN` in here too since it makes more sense that way.

Also title-cased some of the headings and my editor trimmed trailing whitespace of its own accord.

kudos commented:
@guybedford as promised :)

guybedford commented on docs/registries.md:
```diff
@@ -16,16 +16,22 @@ To support private GitHub, simply authenticate with your private GitHub account:
 
 This will enable private repo installs.
 
+#### Github Rate Limiting
```
Perhaps `Continuous Integration - Authentication Tokens`

kudos replied:
I had used `Continuous Integration` first, but this also applies to services like Heroku as well as people just randomly behind proxies and firewalls.

guybedford replied:
Right, I just don't want people to confuse this for normal authentication which should be communicated to users first via `jspm registry config github`. That's also why this should be lower down with a clear title that shows it is specific to automated workflows.

guybedford commented on docs/registries.md:
```diff
@@ -16,16 +16,22 @@ To support private GitHub, simply authenticate with your private GitHub account:
 
+#### Github Rate Limiting
+
+Github rate limits by IP unless an auth token is provided. On hosted services such as Circle, Travis or Heroku where you share IPs with other users also using Github you will likely hit these limits. Setting a `JSPM_GITHUB_AUTH_TOKEN` variable in your environment is enough to tell jspm to use it. Most services have the ability to do this from their web panels.
```
It is worth also noting the `jspm config registries.github.auth AUTH_TOKEN` command.

kudos replied:
Yeah, I think it makes sense to say what happens internally when the env variable is encountered.

guybedford commented on docs/registries.md:
````diff
@@ -119,27 +124,7 @@ For npm, you may wish to automate the loading of config from npmrc. This can be
-before_install:
-- npm install -g jspm
-- jspm config registries.github.auth $JSPM_GITHUB_AUTH_TOKEN
-```
````
I do still like the idea of including direct instructions for users, although perhaps you're right that this is overkill.

kudos replied:
I think given that it's possible to just add environment variables via the web panel now, that is the preferred method. In the scenario most people would encounter, they'll first hit the limits on a remote build, and all they'll need to do is add the env var and not check in any changes to their repo.

guybedford replied:
Sure, that makes sense!

guybedford commented:
I was actually just wondering about it when you sent the PR! Thanks so much for following up, it's really appreciated especially since it is not something that is ever assumed or expected in open source.

kudos commented:
@guybedford I'm a big fan of jspm and systemjs, happy to help where I can :)

kudos commented:
@guybedford I reworded and moved the rate limiting out of registries.md entirely and into installing-packages.md. I also added some SEO juice by including the error message jspm outputs on the doc page.

guybedford commented:
Thanks, I manually merged this in https://github.com/jspm/jspm-cli/commit/6c8752c8d09c8668f90f0b03b39da146a9c683ad, bringing it up further to the top as well.
